mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -370,12 +382,49 @@ pub(super) async fn init_light_client(
 ) -> Result<(AsyncMutex<SaplingSyncConnector>, WalletDbShared), MmError<ZcoinClientInitError>> {
     let (sync_status_notifier, sync_watcher) = channel(1);
     let (on_tx_gen_notifier, on_tx_gen_watcher) = channel(1);
-
-    let lightwalletd_url = Uri::from_str(
-        lightwalletd_urls
-            .first()
-            .or_mm_err(|| ZcoinClientInitError::EmptyLightwalletdUris)?,
-    )?;
+    let mut rpc_clients = Vec::new();
+    let mut errors = Vec::new();
+    if !lightwalletd_urls.is_empty() {
+        for url in lightwalletd_urls {
+            let uri = match Uri::from_str(&*url).map_to_mm(ZcoinClientInitError::InvalidUri) {
+                Ok(uri) => uri,
+                Err(err) => {
+                    errors.push(format!("{:?}", err));
+                    continue;
+                },
+            };
+            let endpoint = match Channel::builder(uri)
+                .tls_config(ClientTlsConfig::new())
+                .map_to_mm(ZcoinClientInitError::TlsConfigFailure)
+            {
+                Ok(endpoint) => endpoint,
+                Err(err) => {
+                    errors.push(format!("{:?}", err));
+                    continue;
+                },
+            };
+            let tonic_channel = match endpoint
+                .connect()
+                .await
+                .map_to_mm(ZcoinClientInitError::ConnectionFailure)
+            {
+                Ok(tonic_channel) => tonic_channel,
+                Err(err) => {
+                    errors.push(format!("{:?}", err));
+                    continue;
+                },
+            };
+            let rpc_copy = CompactTxStreamerClient::new(tonic_channel);
+            rpc_clients.push(rpc_copy);
+        }
+        drop_mutability!(errors);
+        // check if rpc_clients is empty, then for loop wasnt successful
+        if rpc_clients.is_empty() {
+            return Err(format_init_client_e(errors));
+        }
+    } else {
+        return Err(MmError::from(ZcoinClientInitError::EmptyLightwalletdUris));
```

Comment:
Please use early return instead and also note that there are various convenience methods of MmError like `err` mentioned:
```rust
if lightwalletd_urls.is_empty() {
   MmError::err(ZcoinClientInitError::EmptyLightwalletdUris);
}
```
With early return, you can avoid an additional level of nesting of `for` loop.

Reply:
done
added `return MmError::err(ZcoinClientInitError::EmptyLightwalletdUris)` before `for` loop